type-generator/src/main/java/io/mesosphere/types/dtr/Main.java

```diff
@@ -0,0 +1,156 @@
+package io.mesosphere.types.dtr;
+
+import io.mesosphere.types.dtr.compiler.impl.StringTemplateEngine;
+import io.mesosphere.types.dtr.compiler.internal.CompiledFragment;
+import io.mesosphere.types.dtr.compiler.internal.CompilerEngine;
+import io.mesosphere.types.dtr.compiler.internal.CompiledFragmentType;
+import io.mesosphere.types.dtr.compiler.internal.arguments.FileObjectTypeFragmentArguments;
+import io.mesosphere.types.dtr.models.internal.Type;
+import io.mesosphere.types.dtr.models.internal.TypeScope;
+import io.mesosphere.types.dtr.models.internal.TypeScopeRelation;
+import io.mesosphere.types.dtr.models.internal.scopes.FileScope;
+import io.mesosphere.types.dtr.models.internal.scopes.GlobalScope;
+import io.mesosphere.types.dtr.models.internal.types.*;
+import io.mesosphere.types.dtr.models.parser.RAMLParser;
+import io.mesosphere.types.dtr.repository.internal.Repository;
+import io.mesosphere.types.dtr.repository.internal.RepositoryView;
+import org.apache.commons.cli.*;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.List;
+import java.util.Map;
+
+public class Main {
+
+    /**
+     * Entry point
+     * @param args
+     */
+    public static void main(String[] args) throws ParseException {
+        Options options = new Options();
+        CommandLineParser parser = new DefaultParser();
+
+        // Prepare options
+        Option o_help = Option.builder("h")
+                .longOpt("help")
+                .desc("Print this help message")
+                .argName("help")
+                .build();
+
+        Option o_project = Option.builder("p")
+                .longOpt("project")
+                .required()
+                .desc("The project name to target")
+                .hasArg()
+                .argName("project")
+                .build();
+
+        Option o_version = Option.builder("v")
+                .longOpt("version")
+                .required()
+                .desc("The project version to use")
```

Comment:
How are the versions defined? Do we agreed on versioning? I think the new API strives for backwards compatibility. @timcharper has more insight.

Reply:
Check my comment above